db/migrate/20150223125700_create_branches.rb

```diff
@@ -0,0 +1,15 @@
+class CreateBranches < ActiveRecord::Migration
+  def up
+    create_table(:branches) do |t|
+      t.integer :repository_id, null: false
+      t.integer :last_build_id
+      t.string  :name
+      t.timestamps
+    end
+    add_index(:branches, [:repository_id, :name])
```

Comment:
It means that the database won't allow two rows with the same repository_id and branch name. I believe ActiveRecord raises an exception if the unique index is violated, but that depends on `save` vs `save!` if I recall correctly. Either way the save won't "succeed", so we won't have duplicates in our database.